### DISCUSSION TITLE
[ Possible BUG ] Malformed rendering for video component when source changes

### DISCUSSION THREAD

**Discussion by s43** [TheWidlarzGroup/react-native-video#3585] [Q&A]
### What are you trying to do?

I do have a list of videos, and one simple video component that takes a source prop.

Scenario:

* Flash list with a couple of buttons, each represents a video item.
* on each button focus, the source prop of the video component gets updated

Any idea why this occurs? my component is just a simple video component, that works perfectly if there's a stable source passed to it, but once you update this source prop it renders the way I mentioned above.

### What have you tried?

When I do navigate through the FlashList (focus), the selected video shows up but it's totally messed, like on this screenshot:

<img width="514" alt="image" src="https://github.com/react-native-video/react-native-video/assets/4326571/9eab82d8-86b0-40c5-8036-0cd457a3ef7a">

**Comment by freeboub** [comment]
Hey, this is a bug, clearly, I will fix it very soon !

**Comment by freeboub** [comment]
let's close the discussion and follow up in the ticket
